docs/tutorials/upgrading-airbyte.md

```diff
@@ -0,0 +1,89 @@
+_Coming soon! This tutorial is a work in progress for a feature that is currently in development. It is not usable yet!_
+
+# Upgrading Airbyte
+
+## Overview
+
+Airbyte releases a new version of its stable version at least once a week. Some of these releases require updating the data the Airbyte stores internally. This tutorial will describe how to determine if you need to run this ugprade process, and if you do, how to do so. This process does require temporarily turning off Airbyte.
+
+## Determining if you need to Upgrade
+Not every new version will require an upgrade. We follow standard [Semantic Versioning](https://semver.org/) conventions. Data migrations should only be required when upgrading to a new major version of Airbyte. You can always find the latest stable version of Airbyte in our repository [here](https://github.com/airbytehq/airbyte/blob/master/.env#L1). If you are upgrading to a new major version follow the steps below.
```

Comment:
i'm not convinced this is the right way to go, but put it down as a place to start the discussion. my concern is that we are just going to have a lot of major version bumps if we follow this principle and it will look funky.

the other options i see:
1. require migration on every minor version. (this is a bad option since it requires a lot of work on the users part even if the data schema hasn't change)
2. mark in the change log on what minor versions data migration is required. (this isn't great because it's a little easy to miss, but i'm leaning towards this right now, but option 3 is also okay)
3. the option described in the PR following strict semvar convention.

Reply:
Any way we can make airbyte startup fail with a link to the doc if it is not compatible?

Why do you think 1. is bad? I think having a systematic way to upgrade is better than not knowing if you should or not. Keep in mind that this is a new features and we can revise the strategy in the future.




Or for now we just make it a systematic operations

Reply:
The goal of the project is "have a RELIABLE, SAFE and PREDICTABLE way to upgrade", not "have the best UX for upgrade". UX should come later

Reply:
assuming we continue to do 1 minor release per week...

I think there 2 reasons that option 1 isn't great:
1. it's _a lot_ of work for the user. means you're doing this whole migration flow potentially once a week. this migration process already definitely doesn't have the best UX 😆 , but asking someone to do it every week feels like a lot.
2. it means we need to write a migration every week and adds overhead to us doing weekly migrations. though, i guess it's pretty easy to add some syntactic sugar to make no op migrations trivial to write.

that all being said, i'm down to give option 1 a try. none of these options are stellar so we're just picking least bad one. i think i'm persuaded by your framing that option 1 may be least bad.